tests/model/test_model.py

```diff
@@ -2015,6 +2016,46 @@ def test_user_name_from_id_invalid(self, model, user_id):
         with pytest.raises(RuntimeError, match='Invalid user ID.'):
             model.user_name_from_id(user_id)
 
+    @pytest.mark.parametrize('response', [
+        ({
+            "last_event_id": -1,
+            "type": "realm_emoji",
+            "msg": "",
+            "queue_id": "1517975029:0",
+            "realm_emoji": {
+                "100": {
+                    "id": "100",
+                    "name": "urwid",
+                    "deactivated": True
+                },
+                "1": {
+                    "id": "1",
+                    "name": "green_tick",
+                    "deactivated": False
+                },
+                "202020": {
+                    "id": "202020",
+                    "name": "joker",
+                    "deactivated": True
+                },
+            },
+            "result": "success",
+            "zulip_feature_level": 2,
+            "zulip_version": "2.1.0"
+        })
+    ])
+    def test__handle_update_emoji_event(self, mocker, model,
+                                        response):
+        model._handle_update_emoji_event(response)
+        # For an already active emoji which has been deactivated
+        with pytest.raises(KeyError):
+            model.active_emoji_data['urwid']
```

Comment:
Are you trying to achieve the equivalent of `assert "urwid" not in model.active_emoji_data`? If so, that's likely clearer - it's not a specific exception we care about, but whether the key is present (subtle, perhaps, but still).